x-pack/plugins/search_inference_endpoints/public/hooks/use_table_data.tsx

```diff
@@ -27,25 +31,62 @@ interface UseTableDataReturn {
 
 export const useTableData = (
   inferenceEndpoints: InferenceAPIConfigResponse[],
-  queryParams: QueryParams
+  queryParams: QueryParams,
+  filterOptions: FilterOptions,
+  searchKey: string,
+  deploymentStatus: Record<string, DeploymentStatusEnum>
 ): UseTableDataReturn => {
   const tableData: InferenceEndpointUI[] = useMemo(() => {
-    return inferenceEndpoints.map((endpoint) => ({
-      endpoint: endpoint.model_id,
-      provider: endpoint.service,
-      type: endpoint.task_type,
-    }));
-  }, [inferenceEndpoints]);
+    let filteredEndpoints = inferenceEndpoints;
+
+    if (filterOptions.provider.length > 0) {
+      filteredEndpoints = filteredEndpoints.filter((endpoint) =>
+        filterOptions.provider.includes(
+          ServiceProviderKeys[endpoint.service as keyof typeof ServiceProviderKeys]
+        )
+      );
+    }
+
+    if (filterOptions.type.length > 0) {
+      filteredEndpoints = filteredEndpoints.filter((endpoint) =>
+        filterOptions.type.includes(TaskTypes[endpoint.task_type as keyof typeof TaskTypes])
+      );
+    }
+
+    return filteredEndpoints
+      .filter((endpoint) => endpoint.model_id.includes(searchKey))
```

Comment:
why are we doing the search filtering on the frontend? should this be handled in the backend and leverage the ES query?

Reply:
Looks like `GET _inference/_all` does support any `search`, `sort`, or `pagination`. This returns full list of all endpoints.